lib/src/main/java/com/auth0/jwt/oicmsg/KeyBundle.java

```diff
@@ -0,0 +1,519 @@
+    }
+
+    public KeyBundle(String source, String fileFormat, List<String> usage) throws ImportException {
+        this(null, source, 300, true, fileFormat, "RSA", usage);
+    }
+
+    public void doKeys(List<Key> keys) {
+        for (Key keyIndex : keys) {
+            final String kty = keyIndex.getKty();
+            List<String> usage = harmonizeUsage(Arrays.asList(keyIndex.getUse()));
+            keys.remove("use");
+            boolean flag = false;
+            for (String use : usage) {
+                List<String> types = new ArrayList<String>() {{
+                    add(kty);
+                    add(kty.toLowerCase());
+                    add(kty.toUpperCase());
+                }};
+                boolean isSuccess = true;
+                Key key;
+                for (String typeIndex : types) {
+                    try {
+                        switch(typeIndex) {
+                            case "RSA":
+                                key = new RSAKey("use");
+                                break;
+                            case "EC":
+                                key = new ECKey("use");
+                                break;
+                            case "SYMKey":
+                                key = new SYMKey("use");
+                                break;
+                            default:
+                                throw new IllegalArgumentException("Encryption type: " + typeIndex + " isn't supported");
+                        }
+                    } catch (JWKException exception) {
+                        logger.warn("While loading keys: " + exception);
+                        isSuccess = false;
+                    }
+
+                    if (isSuccess) {
+                        this.keys.add(key);
+                        flag = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!flag) {
+                logger.warn("While loading keys, UnknownKeyType: " + kty);
+            }
+        }
+    }
+
+    private static List<String> harmonizeUsage(List<String> uses) {
+        Set<String> keys = map.keySet();
+        Set<String> usagesSet = new HashSet<>();
+        for (String use : uses) {
+            if (keys.contains(use)) {
+                usagesSet.add(use);
+            }
+        }
+        return new ArrayList<>(usagesSet);
+    }
+
+    public void doLocalJwk(String fileName) throws UpdateFailed {
+        JSONParser parser = new JSONParser();
+        try {
+            Object obj = parser.parse(new FileReader(
+                    fileName));
+            JSONObject jsonObject = (JSONObject) obj;
+            JSONArray keys = (JSONArray) jsonObject.get("keys");
+            Iterator<String> iterator = keys.iterator();
+            List<Key> keysList = new ArrayList<Key>();
+            while (iterator.hasNext()) {
+                keysList.add(new Gson().fromJson(iterator.next(), Key.class));
```

Comment:
why?? :( don't create a new instance on each iteration. You can create one at the top of this method or a constants field. Also either GSON or Jackson.

Reply:
good point.  forgot it was in a while :)